src/main/java/de/tum/in/www1/artemis/service/exam/ExamAccessService.java

```diff
@@ -128,7 +129,7 @@ public StudentExam getStudentExamForTestExamElseThrow(Long courseId, Long examId
         Exam exam = studentExam.getExam();
 
         if (!examId.equals(exam.getId())) {
-            throw new AccessForbiddenException("The provided examId does not match with the examId of the studentExam");
+            throw new BadRequestAlertException("The provided examId does not match with the examId of the studentExam", ENTITY_NAME, "examIdMismatch");
```

Comment:
In l. 90, you have used `idMismatch` and here `examIdMismatch` as the error code.